package.py

```diff
@@ -0,0 +1,911 @@
+            f'pyinstaller --noconfirm --clean --distpath "{backend_directory}" rotkehlchen.spec',  # noqa: E501
+            shell=True,
+            env=package_env,
+        )
+        if package_ret_code != 0:
+            logger.error('packaging failed')
+            exit(1)
+        logger.info('Check binary')
+        backend_binary = next(backend_directory.glob(f'**/rotki-core-*-{self.__env.os_tag()}'))
+        ret_code = subprocess.call(f'{backend_binary} version', shell=True)
+        if ret_code != 0:
+            logger.error('backend binary check failed')
+            exit(1)
+
+    @log_group('Pyinstaller')
+    def __install_pyinstaller(self) -> None:
+        if self.__env.is_x86_64() and not self.__env.is_mac_runner():
+            self.pip_install(f'pyinstaller=={pyinstaller_version}')
+        else:
+            self.__build_pyinstaller_bootloader(pyinstaller_version)
+
+
+class FrontendBuilder:
+    def __init__(self, storage: Storage, env: Environment, mac: Optional[MacPackaging]):
+        self.__storage = storage
+        self.__frontend_directory = storage.working_directory.joinpath('frontend')
+        self.__env = env
+        self.__mac = mac
+
+    def __ensure_backend_executable(self) -> None:
+        backend_directory = self.__storage.backend_directory
+        backend_binary = next(backend_directory.glob(f'**/rotki-core-*-{self.__env.os_tag()}'))
+        st = os.stat(backend_binary)
+        os.chmod(backend_binary, st.st_mode | stat.S_IEXEC)
+
+    @staticmethod
+    def check_npm_version() -> None:
+        npm_version = version.parse(subprocess.check_output(
+            'npm --version',
+            encoding='utf-8',
+            shell=True,
+        ))
+        required_version = version.parse('8.0.0')
+        if npm_version < required_version:
+            logger.error(f'The system npm version is < 8.0.0 ({npm_version})')
+            exit(1)
+
+    @log_group('electron app build')
+    def build(self) -> None:
+        self.check_npm_version()
+        self.__storage.check_backend()
+        os.chdir(self.__frontend_directory)
+        if self.__mac is not None:
+            self.__mac.import_signing_certificates()
+
+        frontend_env = self.__env.get_frontend_env()
+
+        self.__restore_npm_dependencies()
+        self.__ensure_backend_executable()
+
+        logger.info('Calling electron:build')
+        ret_code = subprocess.call('npm run electron:build', shell=True, env=frontend_env)
+        if ret_code != 0:
+            logger.error('electron build failed')
+            exit(1)
+
+        frontend_build_dir = self.__frontend_directory.joinpath('app').joinpath('build')
+        os.chdir(frontend_build_dir)
+        for path in frontend_build_dir.iterdir():
+            name = path.name
+            if path.is_dir() or name in ['builder-debug.yml', 'builder-effective-config.yaml']:
+                continue
+
+            suffixes = ['dmg', 'zip', 'exe', 'AppImage', 'tar.xz', 'deb']
```

Comment:
Move this to a constants at the start of the file

Reply:
Why though, what would be the benefit of exposing values only used in this specific function scope to the scope of the whole script?

Reply:
I prefer having this variables that are constant and about the process all together in one place in case they need to be modified to avoid having to go trough the code and finding them when changes are needed